### ISSUE TITLE
Why does it handle object members as strings?

### ISSUE THREAD

**Issue by trilinesoft** [javascript-obfuscator/javascript-obfuscator#351]
## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Obfuscator version used:
* Node version used:

# Stack trace

**Comment by trilinesoft** [comment]
I need all variables and function names to be randomly renamed. But I see that obfuscator doesn't rename member functions of objects and moving them into string array instead and it's quite unacceptable.

Sample code:

`var TestObject = null;

(function () {

    var TestClass = function () {
        var self = this;

        self.PublicFunc = function ()
        {
            var str1 = PrivateFunc1();
            var str2 = PrivateFunc2();

            var text = str1 + str2;

            console.log(text);
            alert(text);
        };

        self.PublicFunc_2 = function ()
        {
            var str1 = PrivateFunc1();
            var str2 = PrivateFunc2();

            var text = str2 + str1;

            console.log(text);
            alert(text);
        }

        function PrivateFunc1() {
            var str = 'hello';
            return str;
        }

        function PrivateFunc2() {
            var str = ' there';
            return str;
        }
    };

    var TestTest = function () {
        alert('test');
    };

    TestObject = new TestClass();

})();

TestObject.PublicFunc();`

Result:

`var a = [
    'PublicFunc',
    'log',
    'PublicFunc_2',
    'hello',
    '\x20there',
    'test'
];
var b = function (c, d) {
    c = c - 0x0;
    var e = a[c];
    return e;
};
var TestObject = null;
(function () {
    var c = function () {
        var d = this;
        d[b('0x0')] = function () {
            var e = k();
            var f = m();
            var g = e + f;
            console[b('0x1')](g);
            alert(g);
        };
        d[b('0x2')] = function () {
            var h = k();
            var i = m();
            var j = i + h;
            console[b('0x1')](j);
            alert(j);
        };
        function k() {
            var l = b('0x3');
            return l;
        }
        function m() {
            var n = b('0x4');
            return n;
        }
    };
    var o = function () {
        alert(b('0x5'));
    };
    TestObject = new c();
}());
TestObject[b('0x0')]();`

PublicFunc() and PublicFunc_2() appeared in the strings list and without any renaming. It is quite inacceptable.

Is it possible to just rename those functions without moving them into the string list?

**Comment by nstarke** [comment]
@trilinesoft is something like this what you mean?
```console.assert = function (){};
console.clear = function (){};
console.count = function (){};
console.error = function (){};
console.group = function (){};
console.groupCollapsed = function (){};
console.groupEnd = function (){};
console.info = function (){};
console.log = function (){};
console.table = function (){};
console.time = function (){};
console.timeEnd = function (){};
console.trace = function (){};
console.warn = function (){};
var 묏쬧柄䍤䙰ƀ渇ﴊ풨ⱖ聉虍ᨍ븴쮀ᣟ넓롘Ѵ獠 = function(糨җ睎, 懢쟊殯ᵉ珼頱佼쵕ᾤ췘ɾ쥖咫輪ઠ矎슷, ۓ受窚琗恭浔횲赦Ṋ꿁樷䌞킼賓ऊ䞟뗳盨, 荏ꃧߤ磾勈핹뀏꼫骛俕鏘㦳ꘇ, 壆, ƈ, ᤂự퀑纏鑑숅) {
    var self = this;

    self.䶑ㆡ쯀澎 = function(岴徯ӌჂ趷ꁥ颴ᚢޱ藉乇扚, 笍纚쳨㽡䟓拗ᴚ㠮瘟穗혳ඛﺡఈ篢欉蚣鲪) {
        var 㔈眥鎵潨䭐ȡ㙴쫲넸퐠䌏읞寚䋁ˢ䦜ᜱ뒳썤 = שׁ㛄ℹ嬬䧃װ츿庠䈼䟇㖂킏礷鲩ꀗ䧮哀鴇ᨬ趃臢쟫藱();
        var 뾂話랂䥆縢쪻㑡悺ᨿߞ嘼ﳮள詝ඓㅵ롽皔 = ᗬޱ眰꼫絥ᇚ敱㬸춘ꕆ彬㩡ㅏ鎩();

        var 秱퉥厣ㅰ뢷 = 㔈眥鎵潨䭐ȡ㙴쫲넸퐠䌏읞寚䋁ˢ䦜ᜱ뒳썤 + 뾂話랂䥆縢쪻㑡悺ᨿߞ嘼ﳮள詝ඓㅵ롽皔;

        console.log(秱퉥厣ㅰ뢷);
        alert(秱퉥厣ㅰ뢷);
    };

    self.㵸嚋ᗷ瀄ￆ損 = function(
        騅쑏踛ꒃ넾懟鞞깚Ɉ뉣瘶,
        乩,
        썩抎聾柹ĩᲄ䮢䳿,
        ᶗ鯜佹뜱Ⴁ陫,
        ᙅ憺ꇼ햚ᰑ䇒媫庄ஈ埡铜幀枈悌읰堿藮,
        庴ኤﺹꡕ哵㻷黋ᢡ岜䖽㹼全횧田裝䰛麪,
        叅솪ꏀ켓㲏鴣翅궰ᣟﺼ퀫ᯘ鲖栨뙼Ꝣᗾאּ䊒,
        ȏ,
        찎꽆족珘硙ܢ噹䒴ꬸꔴ흭鲖ặ鯂鑐쥽付䍁ꔛ了
    ) {
        var 㔈眥鎵潨䭐ȡ㙴쫲넸퐠䌏읞寚䋁ˢ䦜ᜱ뒳썤 = שׁ㛄ℹ嬬䧃װ츿庠䈼䟇㖂킏礷鲩ꀗ䧮哀鴇ᨬ趃臢쟫藱();
        var 뾂話랂䥆縢쪻㑡悺ᨿߞ嘼ﳮள詝ඓㅵ롽皔 = ᗬޱ眰꼫絥ᇚ敱㬸춘ꕆ彬㩡ㅏ鎩();

        var 秱퉥厣ㅰ뢷 = 뾂話랂䥆縢쪻㑡悺ᨿߞ嘼ﳮள詝ඓㅵ롽皔 + 㔈眥鎵潨䭐ȡ㙴쫲넸퐠䌏읞寚䋁ˢ䦜ᜱ뒳썤;

        console.log(秱퉥厣ㅰ뢷);
        alert(秱퉥厣ㅰ뢷);
    }

    function שׁ㛄ℹ嬬䧃װ츿庠䈼䟇㖂킏礷鲩ꀗ䧮哀鴇ᨬ趃臢쟫藱(
        爀몆ꣲ屮郈優뛊튢䦶뜇逩쀁蒜ކ,
        䧵,
        靟ᰋぶ죕ꭅ㜲蟯俍ര,
        딑㝟狼琬럜夶叐刯곫즪䒏ꕑ揵鹘Ĩ䗹,
        衴Ⴆ밈ᆹ,
        澿뀬,
        瘝坭儍樥㪔쵃賮玲얅ꂼႳ鞕詆,
        훯괂谆,
        留流坢ⶋ蓹ڢ䠮ꮣ筍䵨ⱪ,
        콺掰뺍ꅳ㭓엃䈞툟虙ԭ쬪浓廀ᮻ,
        ೱ緧죥㛳쟞뒬劋宁㑩
    ) {
        var 㳥 = 'hello';
        return 㳥;
    }

    function ᗬޱ眰꼫絥ᇚ敱㬸춘ꕆ彬㩡ㅏ鎩(
        鹝匣廯貤戇㨔쾠赵堻夫闾儇ꃤ齯伓榧硍ⱎ,
        녏膘磣몶嗡,
        覈䓣㤿뵮鄼酯列䊗秶첀鬂䉪仓栈,
        Ɲ沈暑핪ᙽ峪驎鑖胫펑,
        酽鎸鯳梨䂗ꆪ玤䥗,
        箌䠺툊엚ȷ疙ꊘ䃍㵙䬸璺茵䏊濹,
        Ἶ敻䛆늉鬍讎瞇㵽,
        姁㵦瘮儰飾㟦瓻ꌾ繜饹蓵푀ꐕᦚ룬鳖ꉐ믊ᝣ鈀癭㵝볐
    ) {
        var 㳥 = ' there';
        return 㳥;
    }
};

var ꝟ定㱔㿽압䨲馈紺셭앧ľ剣膣긒詐龮浥럺꼫碲嶨뢿髈曋 = function(
    懍즻쓜ന猲懝득呁흝㨺紶㴾,
    悉禼䔁献ඓꏦ沊졌휸,
    徛莿瘮伽葯臃栗밯죶箖啟䝛긠ᮎ鲞쭶캭钿ᛚꩠ笐,
    ﳞ鲕찥썇栲譁鯟켷,
    庄퉟幔臖Ⴎ硸桬꽇朵荐峌닛頜㖂ᐅᚖ춌벽舍,
    乗ꗗ챾뻞쮚泰ꃳ塿鱉ꞵ귽咇쉸궘痜돐
) {
    alert('test');
};

礼鬍澿刕ヵ츷냯駹㣐 = new 묏쬧柄䍤䙰ƀ渇ﴊ풨ⱖ聉虍ᨍ븴쮀ᣟ넓롘Ѵ獠();

```

**Comment by zamotkin** [comment]
It's not a bug. It will be realized in future versions of obfuscator.

**Comment by sanex3339** [comment]
Mangle prop discussed here:
https://github.com/javascript-obfuscator/javascript-obfuscator/issues/270

**Comment by sanex3339** [comment]
I implemented it here: 
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/625

**Comment by sanex3339** [comment]
`renameProperties` option released as `1.1.0` version.
